src/storage/store/rel_table.cpp

```diff
@@ -164,63 +155,72 @@ bool RelTable::delete_(Transaction* transaction, TableDeleteState& deleteState)
     return bwdDeleted;
 }
 
-void RelTable::detachDelete(Transaction* transaction, RelDataDirection direction,
-    ValueVector* srcNodeIDVector, RelDetachDeleteState* deleteState) {
-    KU_ASSERT(srcNodeIDVector->state->getSelVector().getSelSize() == 1);
+void RelTable::detachDelete(Transaction* transaction, RelDataDirection direction, 
+    RelTableDeleteState* deleteState) {
+    KU_ASSERT(deleteState->srcNodeIDVector.state->getSelVector().getSelSize() == 1);
     const auto tableData =
         direction == RelDataDirection::FWD ? fwdRelTableData.get() : bwdRelTableData.get();
     const auto reverseTableData =
         direction == RelDataDirection::FWD ? bwdRelTableData.get() : fwdRelTableData.get();
     std::vector<column_id_t> columnsToScan = {NBR_ID_COLUMN_ID, REL_ID_COLUMN_ID};
-    const auto relIDVectors = std::vector<ValueVector*>{deleteState->dstNodeIDVector.get(),
-        deleteState->relIDVector.get()};
+    const auto relIDVectors = std::vector<ValueVector*>{&deleteState->dstNodeIDVector,
```

Comment:
`relIDVectors` doesn't seem to be an accurate name. I think you can directly initialize vectors into `relReadState->outputVectors`?

Reply:
sure